build/constants.yaml

```diff
@@ -174,3 +174,30 @@
     - type: shell
       directory: build
       file-name-prefix: __
+- name: detector-nuget-packages-versions
```

Comment:
why not have this just once, but an array of different outputs?

Reply:
Yeah I had tried it, but the problem was with how the 'values' of these keys are being interpreted by the scripts. For example, I want to use `pre-$(BUILD_BUILDNUMBER)` for msbuild scenario, but since this content also gets generated for `bash` file, it throws an error saying that it is invalid to use `$(BUILD_BUILDNUMBER)`, so I had to split it into multiple sections.